### PR TITLE
MC-9453: Load balancer: update list/retrieve docs, add delete operation

### PR DESCRIPTION
**Pull request by SeventhOfNine**
Update:
- list lb
- retrieve lb

Add:
- delete lb

### CODE REVIEW

File: source/includes/gcp/_load_balancer.md [hunk 1]
```diff
@@ -48,18 +50,107 @@ Retrieve a list of all load balancers in a given [environment](#administration-e
...
+If a health check is specified, the backend using this health check in the load balancer, will also be selected for deletion.
+</aside>
 
+<aside class="notice">
+Any specified optional resources to be removed will be deleted if they can be deleted. If any other resource outside of the current load balancer is using them, the delete will not happen. The overall delete operation will still be marked as successful.
+</aside>
 
+Optional | &nbsp;
+------ | -----------
+`shortBackendsToDelete`<br/>*List<String>* | A list of backend services to delete
```
Comment (pdube) [inline source/includes/gcp/_load_balancer.md:166]: The doc says it should be a list of String, but the example shows just a String?

File: source/includes/gcp/_load_balancer.md [hunk 2]
```diff
@@ -48,18 +50,107 @@ Retrieve a list of all load balancers in a given [environment](#administration-e
...
+        ],
+        "urlMap": "https://www.googleapis.com/compute/v1/projects/test-area/global/urlMaps/um-test-dfr",
+        "backendServices": [
+            "https://www.googleapis.com/compute/v1/projects/test-area/global/backendServices/be-test-jik"
+        ],
+        "name": "um-test-dfr",
+        "protocol": "HTTPS",
+        "publicAddress": "34.98.108.48, 34.102.228.10",
+        "reserveStaticIP": false,
+        "shortDisplayBackends": "be-test-jik"
```
Comment (pdube) [inline source/includes/gcp/_load_balancer.md]: Is this only for UI? Should we remove this from the API doc?

File: source/includes/gcp/_load_balancer.md [hunk 3]
```diff
@@ -48,18 +50,107 @@ Retrieve a list of all load balancers in a given [environment](#administration-e
...
+`forwardingRules`<br/>*List<String>* | The list of forwarding rules for this resource
+`id`<br/>*String* | The id of this resource, which should be the same id as corresponding Url Map
 `name`<br/>*String* | The name of this resource
-`protocol`<br/>*String* | The protocol of this resource
-`publicAddress`<br/>*String* | The public address which allow user access their service quickly
-`shortBackends`<br/>*String* | The list of backends for this resource in string format 
-`targetProxy`<br/>*String* | Server-defined URL for corresponding target proxy
+`protocol`<br/>*String* | The protocol of this resource. If there are multiple target proxies, and at least one of them is HTTPS, the protocol for the load balancer is set to HTTPS.
+`publicAddress`<br/>*List<String>* | The list of external addresses
+`reserveStaticIP`<br/>*boolean* | [Read-only] The type of external IP used, false = ephemeral, true = static
```
Comment (pdube) [inline source/includes/gcp/_load_balancer.md]: Is this for the operation?

File: source/includes/gcp/_load_balancer.md [hunk 4]
```diff
@@ -18,22 +18,24 @@ curl -X GET \
...
+                "https://www.googleapis.com/compute/v1/projects/test-area/global/targetHttpsProxies/tp-test-rqi"
+            ],
+            "urlMap": "https://www.googleapis.com/compute/v1/projects/test-area/global/urlMaps/um-test-dfr",
+            "backendServices": [
+                "https://www.googleapis.com/compute/v1/projects/test-area/global/backendServices/be-test-jik"
             ],
-            "shortBackends": "instance-group-1, instance-group-2"
+            "name": "um-test-dfr",
+            "protocol": "HTTPS",
+            "publicAddress": "34.98.108.48, 34.102.228.10",
```
Comment (pdube) [inline source/includes/gcp/_load_balancer.md]: This is a list?

File: source/includes/gcp/_load_balancer.md (hunk 1 above)
Comment (SeventhOfNine) [reply source/includes/gcp/_load_balancer.md:166]: I'll update the examples

File: source/includes/gcp/_load_balancer.md (hunk 2 above)
Comment (SeventhOfNine) [reply source/includes/gcp/_load_balancer.md]: Yes this is only for the list view

File: source/includes/gcp/_load_balancer.md (hunk 3 above)
Comment (SeventhOfNine) [reply source/includes/gcp/_load_balancer.md]: No, I will remove

File: source/includes/gcp/_load_balancer.md (hunk 4 above)
Comment (SeventhOfNine) [reply source/includes/gcp/_load_balancer.md]: This is not for the delete operation, it's used in create / list. I will remove

File: source/includes/gcp/_load_balancer.md [hunk 5]
```diff
@@ -48,18 +48,101 @@ Retrieve a list of all load balancers in a given [environment](#administration-e
...
+`backendServices`<br/>*List<String>* | The list of backends for this resource
+`forwardingRules`<br/>*List<String>* | The list of forwarding rules for this resource
+`id`<br/>*String* | The id of this resource, which should be the same id as corresponding Url Map
 `name`<br/>*String* | The name of this resource
-`protocol`<br/>*String* | The protocol of this resource
-`publicAddress`<br/>*String* | The public address which allow user access their service quickly
-`shortBackends`<br/>*String* | The list of backends for this resource in string format 
-`targetProxy`<br/>*String* | Server-defined URL for corresponding target proxy
+`protocol`<br/>*String* | The protocol of this resource. If there are multiple target proxies, and at least one of them is HTTPS, the protocol for the load balancer is set to HTTPS. Valid values are HTTP or HTTPS.
+`publicAddress`<br/>*List<String>* | The list of external addresses
```
Comment (JamieWulala) [inline source/includes/gcp/_load_balancer.md]: maybe plural since it's multiple addresses?

File: source/includes/gcp/_load_balancer.md [hunk 6]
```diff
@@ -48,18 +48,101 @@ Retrieve a list of all load balancers in a given [environment](#administration-e
...
+            "https://www.googleapis.com/compute/v1/projects/test-area/global/targetHttpProxies/tp-test-fkb",
+            "https://www.googleapis.com/compute/v1/projects/test-area/global/targetHttpsProxies/tp-test-rqi"
+        ],
+        "urlMap": "https://www.googleapis.com/compute/v1/projects/test-area/global/urlMaps/um-test-dfr",
+        "backendServices": [
+            "https://www.googleapis.com/compute/v1/projects/test-area/global/backendServices/be-test-jik"
+        ],
+        "name": "um-test-dfr",
+        "protocol": "HTTPS",
+        "publicAddress": "34.98.108.48, 34.102.228.10"
```
Comment (JamieWulala) [inline source/includes/gcp/_load_balancer.md]: here too plural?

File: source/includes/gcp/_load_balancer.md [hunk 7]
```diff
@@ -48,18 +48,101 @@ Retrieve a list of all load balancers in a given [environment](#administration-e
...
+Retrieve a load balancer in a given [environment](#administration-environments)
+
+Attributes | &nbsp;
+------- | -----------
+`backendServices`<br/>*List<String>* | The list of backends for this resource
+`forwardingRules`<br/>*List<String>* | The list of forwarding rules for this resource
+`id`<br/>*String* | The id of this resource, which should be the same id as corresponding Url Map
+`name`<br/>*String* | The name of this resource
+`protocol`<br/>*String* | The protocol of this resource. If there are multiple target proxies, and at least one of them is HTTPS, the protocol for the load balancer is set to HTTPS. Valid values are HTTP or HTTPS.
+`publicAddress`<br/>*List<String>* | The list of external addresses
```
Comment (JamieWulala) [inline source/includes/gcp/_load_balancer.md]: plural

File: source/includes/gcp/_load_balancer.md (hunk 1 above)
Comment (SeventhOfNine) [reply source/includes/gcp/_load_balancer.md:166]: Done

File: source/includes/gcp/_load_balancer.md (hunk 2 above)
Comment (SeventhOfNine) [reply source/includes/gcp/_load_balancer.md]: Done

File: source/includes/gcp/_load_balancer.md (hunk 3 above)
Comment (SeventhOfNine) [reply source/includes/gcp/_load_balancer.md]: Done

File: source/includes/gcp/_load_balancer.md (hunk 4 above)
Comment (SeventhOfNine) [reply source/includes/gcp/_load_balancer.md]: Renamed to `publicAddresses`

File: source/includes/gcp/_load_balancer.md (hunk 5 above)
Comment (SeventhOfNine) [reply source/includes/gcp/_load_balancer.md]: renamed to `publicAddresses`

File: source/includes/gcp/_load_balancer.md (hunk 6 above)
Comment (SeventhOfNine) [reply source/includes/gcp/_load_balancer.md]: fixed

File: source/includes/gcp/_load_balancer.md (hunk 7 above)
Comment (SeventhOfNine) [reply source/includes/gcp/_load_balancer.md]: fixed

File: source/includes/gcp/_load_balancer.md (hunk 3 above)
Comment (SeventhOfNine) [reply source/includes/gcp/_load_balancer.md]: This comes from create operation. I removed the description of it

**Review by JamieWulala — APPROVED** [review]
LGTM!

File: source/includes/gcp/_load_balancer.md [hunk 8]
```diff
@@ -18,22 +18,22 @@ curl -X GET \
...
+                "https://www.googleapis.com/compute/v1/projects/test-area/global/targetHttpsProxies/tp-test-rqi"
+            ],
+            "urlMap": "https://www.googleapis.com/compute/v1/projects/test-area/global/urlMaps/um-test-dfr",
+            "backendServices": [
+                "https://www.googleapis.com/compute/v1/projects/test-area/global/backendServices/be-test-jik"
             ],
-            "shortBackends": "instance-group-1, instance-group-2"
+            "name": "um-test-dfr",
+            "protocol": "HTTPS",
+            "publicAddresses": "34.98.108.48, 34.102.228.10"
```
Comment (pdube) [inline source/includes/gcp/_load_balancer.md]: If this is only for the UI, we should not be exposing it IMO

File: source/includes/gcp/_load_balancer.md [hunk 9]
```diff
@@ -48,18 +48,101 @@ Retrieve a list of all load balancers in a given [environment](#administration-e
...
+
+```shell
+curl -X DELETE \
+   -H "MC-Api-Key: your_api_key" \
+   "https://cloudmc_endpoint/v1/services/gcp/test-area/loadbalancers/8268558443601303519"
+```
+
+```json
+{
+  "shortBackendsToDelete": "be-test-jik, be-other-srd",
```
Comment (pdube) [inline source/includes/gcp/_load_balancer.md]: Shouldn't this be `"shortBackendsToDelete":[ "be-test-jik", "be-other-srd"]` ?

File: source/includes/gcp/_load_balancer.md (hunk 8 above)
Comment (SeventhOfNine) [reply source/includes/gcp/_load_balancer.md]: Can we tag fields like that not to be visible through the API ?
I can remove from the docs, no problem. But anyone using the API will see the fields.

File: source/includes/gcp/_load_balancer.md (hunk 9 above)
Comment (SeventhOfNine) [reply source/includes/gcp/_load_balancer.md]: <run/>

Comment (SeventhOfNine) [reply source/includes/gcp/_load_balancer.md]: Fixed

File: source/includes/gcp/_load_balancer.md (hunk 8 above)
Comment (SeventhOfNine) [reply source/includes/gcp/_load_balancer.md]: Removed from doc

**Review by pdube — APPROVED** [review]

**Pull request merged**